### PR TITLE
Fix Notice in RuleEvaluator

### LINKED ISSUE #7825: (WC 5.9 Beta) PHP Notice/Warning for RemoteInboxNotifications/RuleEvaluator.php upon activating WC
### Describe the bug

When activating WC 5.9 Beta 1 there is a PHP Notice/Warning depending on the PHP Version

For Atomic PHP 7.4 Sites this PHP notice is shown

```
NOTICE: /srv/htdocs/wp-content/plugins/woocommerce/packages/woocommerce-admin/src/RemoteInboxNotifications/RuleEvaluator.php:61 - Trying to get property 'type' of non-object
set_current_screen, WP_Screen->set_current_screen, do_action('current_screen'), WP_Hook->do_action, WP_Hook->apply_filters, Automattic\WooCommerce\Admin\Features\Onboarding::redirect_wccom_install, Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskLists::get_list, Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskLists::get_lists, Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskLists::maybe_add_default_tasks, Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks\Marketing::get_task, Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks\Marketing::get_bundles, Automattic\WooCommerce\Admin\Features\RemoteFreeExtensions\Init::get_extensions, Automattic\WooCommerce\Admin\Features\RemoteFreeExtensions\EvaluateExtension::evaluate, Automattic\WooCommerce\Admin\RemoteInboxNotifications\RuleEvaluator->evaluate
```

For Atomic PHP 8.0 Sites this PHP warning is shown
```
WARNING: /srv/htdocs/wp-content/plugins/woocommerce/packages/woocommerce-admin/src/RemoteInboxNotifications/RuleEvaluator.php:61 - Attempt to read property "type" on bool
set_current_screen, WP_Screen->set_current_screen, do_action('current_screen'), WP_Hook->do_action, WP_Hook->apply_filters, Automattic\WooCommerce\Admin\Features\Onboarding::redirect_wccom_install, Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskLists::get_list, Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskLists::get_lists, Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskLists::maybe_add_default_tasks, Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks\Marketing::get_task, Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks\Marketing::get_bundles, Automattic\WooCommerce\Admin\Features\RemoteFreeExtensions\Init::get_extensions, Automattic\WooCommerce\Admin\Features\RemoteFreeExtensions\EvaluateExtension::evaluate, Automattic\WooCommerce\Admin\RemoteInboxNotifications\RuleEvaluator->evaluate
```

I have confirmed this doesn't happen with WooCommerce 5.8

### Additional context
No other plugins active. I haven't seen this on any other hosts yet.

System Status Report

<details>

`
### WordPress Environment ###

WordPress address (URL): https://ephemeral-tammullen-20211020.atomicsites.blog
Site address (URL): https://ephemeral-tammullen-20211020.atomicsites.blog
WC Version: 5.9.0
REST API Version: ✔ 5.9.0
WC Blocks Version: ✔ 6.1.0
Action Scheduler Version: ✔ 3.3.0
WC Admin Version: ✔ 2.8.0-rc.2
Log Directory Writable: ✔
WP Version: 5.8.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.0.11
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.78.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.4.20-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.9.0
WC Database Prefix: wp_
Total Database Size: 2.47MB
Database Data Size: 1.07MB
Database Index Size: 1.40MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.13MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 0.20MB + Index: 0.03MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 7
post: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 5.9.0-beta.1

### Inactive Plugins (2) ###

Akismet Anti-Spam: by Automattic – 4.2.1
Jetpack: by Automattic – 10.3-a.3

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-One
Version: 1.4
Author URL: https://wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 1
Oldest: 2021-10-20 11:03:41 +0000
Newest: 2021-10-20 11:03:41 +0000

Pending: 1
Oldest: 2021-10-20 14:05:48 +0000
Newest: 2021-10-20 14:05:48 +0000

### Status report information ###

Generated at: 2021-10-20 14:04:59 +00:00

`
</details>

### PR DESCRIPTION
**Pull request by Mte90**
Fixes #7825 

no changelog

### CODE REVIEW

**Comment by moon0326** [comment]
Thank you for opening this PR @Mte90 

I've looked at https://github.com/woocommerce/woocommerce-admin/issues/7825, but I can't reproduce the error anymore.

Since you're having errors on your setup, could you kindly share the steps to reproduce the errors so that I can test this fix locally?

The fix looks simple and should fix non-object related issues.

File: src/RemoteInboxNotifications/RuleEvaluator.php [hunk 1]
```diff
@@ -63,6 +63,10 @@ public function evaluate( $rules, $stored_state = null, $logger_args = array() )
 		}
 
 		foreach ( $rules as $rule ) {
+			if ( !is_object( $rule ) ) {
```
Comment (moon0326) [inline src/RemoteInboxNotifications/RuleEvaluator.php:66]: I'm seeing failed `Lint the PHP` job. You can see the errors on https://github.com/woocommerce/woocommerce-admin/runs/4876120052?check_suite_focus=true.

I'm not sure how it passed the lint pre-hook locally.

You can fix the style automatically by running the following command.

`./vendor/bin/phpcbf --standard=phpcs.xml.dist src/RemoteInboxNotifications/RuleEvaluator.php`

Comment (Mte90) [reply src/RemoteInboxNotifications/RuleEvaluator.php:66]: I did just a c&p from the Github UI...

**Comment by Mte90** [comment]
About the error I can say that with this fix I don't have anymore the error (with the latest release 6.1.0). I don't know how to replicate the error locally as I have a production website where is happening with 30 plugins.

File: src/RemoteInboxNotifications/RuleEvaluator.php (hunk 1 above)
Comment (moon0326) [reply src/RemoteInboxNotifications/RuleEvaluator.php:66]: We still need to fix the style errors as it is a required task to merge.

```
 66 | ERROR | [x] Expected 1 space after "!"; 0 found
 69 | ERROR | [x] Whitespace found at end of line
```

**Comment by Mte90** [comment]
I should be able to fix the lint errors

**Review by moon0326 — APPROVED** [review]
Thank you for the fix @Mte90 

It LGTM 👍

**Comment by Mte90** [comment]
When it is planned the merge? I am still patching manually the file on every woocommerce release.

**Comment by chihsuan** [comment]
> When it is planned the merge? I am still patching manually the file on every woocommerce release.

@moon0326 Are we ready to merge this?

**Comment by moon0326** [comment]
@Mte90 @chihsuan 
My apologies. Yes, this is ready to go.

**Pull request merged**